Part 3 working with raster data

cforgaci opened the pull request.

manuGil reviewed (approved):
Please consider my comments to improve this lessons.

ClementineCttn commented on episodes/09-open-and-plot-vector-layers.Rmd:
```diff
@@ -0,0 +1,189 @@
+2. What is the CRS and extent for each object?
+3. Do the files contain points, lines, or polygons?
+4. How many features are in each file?
+
```
1 star for each answer

ClementineCttn reviewed (approved):
Hello. This lesson is still quite long and a bit abstract. I think it might help to skip a few operations and instead give some context on why we might want to subtract raster or what to do with the result (in the context of urban and social research and practice)

cforgaci commented on episodes/14-plot-raster-data.Rmd:
```diff
@@ -0,0 +1,289 @@
+
+:::
+
+## Layering Rasters
```
I changed the first paragraph according to your suggestion. I kept layering in the title though. I think it works there, and it remains consistent with the original lesson.

cforgaci commented on episodes/16-raster-calculations.Rmd:
```diff
@@ -0,0 +1,188 @@
+
+# Things you'll need to complete this episode
+
+See the [lesson homepage]() for detailed information about the software, data, and other prerequisites you will need to work through the examples in this episode.
```
```suggestion
See the [lesson homepage](../index.md) for detailed information about the software, data, and other prerequisites you will need to work through the examples in this episode.
```

cforgaci commented on episodes/16-raster-calculations.Rmd:
```diff
@@ -0,0 +1,188 @@
+
+:::
+
+We often want to combine values of and perform calculations on rasters to create a new output raster. This episode covers how to subtract one raster from another using basic raster math and the `lapp()` function. It also covers how to extract pixel values from a set of locations - for example a buffer region around plot locations at a field site.
```
```suggestion
We often want to combine values of and perform calculations on rasters to create a new output raster. This episode covers how to subtract one raster from another using basic raster math and the `lapp()` function. For instance, this operation is useful to obtain building and tree heights by subtracting a raster that contains bare terrain (DTM) from a raster containing the terrain with heights of buildings and trees included (DSM). 
```

cforgaci commented on episodes/16-raster-calculations.Rmd:
```diff
@@ -0,0 +1,188 @@
+<!-- This lesson uses the `terra` package in particular. If you have not installed it yet, do so by running `install.packages("terra")` before loading it with `library(terra)`. -->
+
+:::
+
```
I added an example and removed the last sentence (a stray sentence from the original lesson). See below.

cforgaci commented on episodes/16-raster-calculations.Rmd:
```diff
@@ -0,0 +1,188 @@
+
+We often want to perform calculations on two or more rasters to create a new output raster. For example, if we are interested in mapping the heights of trees and buildings across an entire field site, we might want to calculate the difference between the Digital Surface Model (DSM, tops of trees and buildings) and the Digital Terrain Model (DTM, ground level). The resulting dataset is referred to ...
+
+![Source: National Ecological Observatory Network (NEON).](https://datacarpentry.org/r-raster-vector-geospatial/fig/dc-spatial-raster/lidarTree-height.png)
```
Sorry, I don't understand the comment.

manuGil replied:
Hello @cforgaci , My comment is about this figure in the lesson:

![](https://datacarpentry.org/r-raster-vector-geospatial/fig/dc-spatial-raster/lidarTree-height.png)

The top-left figure is similar (almost the same) to the one on the bottom (CHM), and that is confusing by looking into the figure only. I understand the explanation on the text but it is not evident in the figure.

cforgaci replied:
Thanks for pointing this out. You're right. It does remove the elevation in the latter, but it is not clear in this figure. I'll see if I can find an alternative, maybe including buildings too to fit our lesson better.

cforgaci commented on episodes/16-raster-calculations.Rmd:
```diff
@@ -0,0 +1,188 @@
+
+## Raster math and Canopy Height Models
+
+We can perform raster calculations by subtracting (or adding, multiplying, etc) two rasters. In the geospatial world, we call this “raster math”.
```
```suggestion
We can perform raster calculations by subtracting (or adding, multiplying, etc) two rasters. In the geospatial world, we call this “raster math” or "map algebra".
```

cforgaci commented on episodes/16-raster-calculations.Rmd:
````diff
@@ -0,0 +1,188 @@
+     coord_quickmap()
+```
+
+## Raster math and Canopy Height Models
````
I added this to the first sentence. I kept the title for consistency with the original lesson.

cforgaci commented on episodes/13-intro-to-raster-data.Rmd:
```diff
@@ -0,0 +1,268 @@
+
+# Things you'll need to complete this episode
+
+See the [lesson homepage]() for detailed information about the software, data, and other prerequisites you will need to work through the examples in this episode.
```
```suggestion
See the [lesson homepage](../index.md) for detailed information about the software, data, and other prerequisites you will need to work through the examples in this episode.
```

cforgaci commented on episodes/13-intro-to-raster-data.Rmd:
````diff
@@ -0,0 +1,268 @@
+summary(values(DSM_TUD))
+```
+
+To visualise this data in R using `ggplot2`, we need to convert it to a data frame. We learned about data frames in an [earlier lesson](). The `terra` package has a built-in function for conversion to a data frame.
````
```suggestion
To visualise this data in R using `ggplot2`, we need to convert it to a data frame. We learned about data frames in an [earlier lesson](../episodes/03-explore-data.Rmd). The `terra` package has a built-in function for conversion to a data frame.
```

cforgaci commented on episodes/14-plot-raster-data.Rmd:
````diff
@@ -0,0 +1,289 @@
+    scale_fill_manual(values = terrain.colors(3)) + 
+    coord_quickmap()
+```
+## More Plot Formatting
````
Done

cforgaci commented on episodes/14-plot-raster-data.Rmd:
done with `fig.width = 12` on line 12

cforgaci commented on episodes/14-plot-raster-data.Rmd:
````diff
@@ -0,0 +1,289 @@
+  scale_alpha(range = c(0.15, 0.65), guide = "none") +  
+  ggtitle("Elevation with hillshade") +
+  coord_quickmap()
+```
````
I changed the colour scale to `H`, called "turbo", which gives more contrast. Thanks for pointing this out.

cforgaci commented on episodes/14-plot-raster-data.Rmd:
````diff
@@ -0,0 +1,289 @@
+
+To see the cut-off values for the groups, we can ask for the unique values of `fct_elevation`:
+```{r}
+unique(DSM_TUD_df$fct_elevation)
````
I replaced all occurrences of `unique()` with `levels()`

cforgaci commented:
Hi, @ClementineCttn. Thanks for the comments.

> Hello. This lesson is still quite long and a bit abstract. I think it might help to skip a few operations 

- [ ] Drop a few operations to shorten raster lesson 

> and instead give some context on why we might want to subtract raster or what to do with the result (in the context of urban and social research and practice)

I added some text to explain this. @manuGil also suggested that the figure explaining the CHM is not clear. I'll try to search for one that is closer to urbanism applications.

- [ ] Replace CHM figure

Selkubi commented on episodes/13-intro-to-raster-data.Rmd:
```diff
@@ -0,0 +1,269 @@
+3. How large would a 5x5 pixel area be on the Earth’s surface?
+4. Is the file a multi- or single-band raster?
+
+Note that this file is a hillshade raster. We will learn about hillshades in the [Working with Multi-band Rasters in R episode.
```
```suggestion
Note that this file is a hillshade raster. We will learn about hillshades in the [Working with Multi-band Rasters in R](../episodes/17-work-with-multi-band-rasters.Rmd) episode.
```

Selkubi commented on episodes/14-plot-raster-data.Rmd:
```diff
@@ -0,0 +1,291 @@
+2. Axis labels.
+3. A plot title.
+
+::: solution
```
The solution starts on top of the '3. A plot title' line for me
<img width="445" alt="Capture" src="https://github.com/carpentries-incubator/r-geospatial-urban/assets/63798892/0601f4bc-5b29-499b-9886-d48e2d6d79e3">
.

cforgaci replied:
Thanks for pointing this out, @Selkubi. The issue was that the callout label was `Challenge`, upper case, instead of `challenge`. It should work now.

cforgaci commented on episodes/15-reproject-raster-data.Rmd:
```diff
@@ -0,0 +1,223 @@
+                 aes(x = x, y = y, 
+                   alpha = `tud-dtm-5m-hill-ETRS89`)) +
+     scale_fill_gradientn(name = "Elevation", colors = terrain.colors(10)) + 
+     coord_quickmap()
```
```suggestion
     coord_equal()
```

dcannatella commented on episodes/13-intro-to-raster-data.Rmd:
````diff
@@ -0,0 +1,269 @@
+```
+
+::: questions 
+- What is a raster dataset?
````
good question

dcannatella reviewed (commented):
check it out!

dcannatella commented on episodes/13-intro-to-raster-data.Rmd:
```diff
@@ -0,0 +1,269 @@
+
+:::
+
+In this lesson we will work with raster data. We will start with an introduction of the fundamental principles and metadata needed to work with raster data in R. We will discuss some of the core metadata elements needed to understand raster data in R, including CRS and resolution. 
```
```suggestion
In this lesson, we will work with raster data. We will start with an introduction of the fundamental principles and metadata needed to work with raster data in R. We will discuss some of the core metadata elements needed to understand raster data in R, including CRS and resolution. 
```

dcannatella commented on episodes/13-intro-to-raster-data.Rmd:
````diff
@@ -0,0 +1,277 @@
+DSM_TUD <- rast("data/tud-dsm-5m.tif")
+DSM_TUD
+```
+The information above includes a report on dimension, resulotion, extent and CRS, but no information about the values. Similar to other data structures in R like vectors and data frame columns, descriptive statistics for raster data can be retrieved with the `summary()` function.
````
```suggestion
The information above includes a report on dimension, resolution, extent and CRS, but no information about the values. Similar to other data structures in R like vectors and data frame columns, descriptive statistics for raster data can be retrieved with the `summary()` function.
```

dcannatella replied:
```suggestion
The information above includes a report on dimension, resolution, extent and CRS, but no information about the values. Similar to other data structures in R like vectors and data frame columns, descriptive statistics for raster data can be retrieved with the `summary()` function.
```

dcannatella commented on episodes/13-intro-to-raster-data.Rmd:
```diff
@@ -0,0 +1,277 @@
+
+::: solution
+
+`+units=m` in the output of the code above tells us that our data is in meters.
```
```suggestion
`+units=m` in the output of the code above tells us that our data is in meters (m).
```

dcannatella reviewed (commented):
Review of Episode 13

dcannatella commented on episodes/14-plot-raster-data.Rmd:
```diff
@@ -0,0 +1,295 @@
+DTM_hill_TUD <- rast("data/tud-dtm-5m-hill.tif")
+DTM_hill_TUD_df <- as.data.frame(DTM_hill_TUD, xy = TRUE)
+
+ggplot() +
```
the final plot looks quite grey... is it ok?

cforgaci replied:
I know it is. Grey is typically used for hillshade... not that useful in our case. I am considering dropping this part as it is not very relevant for us.

cforgaci replied:
I removed that part.

dcannatella reviewed (commented):
review episode 15 done

dcannatella reviewed (commented):
episode 15 reviewed

dcannatella commented on episodes/17-work-with-multi-band-rasters.Rmd:
```diff
@@ -0,0 +1,169 @@
+
+## Getting Started with Multi-Band Data in R
+
+In this episode, the multi-band data that we are working with is Beeldmateriaal Open Data from the Netherlands. Each RGB image is a 3-band raster. The same steps would apply to working with a multi-spectral image with 4 or more bands - like Landsat imagery.
```
```suggestion
In this episode, the multi-band data that we are working with is [Beeldmateriaal Open Data] (https://opendata.beeldmateriaal.nl/) from the Netherlands. Each RGB image is a 3-band raster. The same steps would apply to working with a multi-spectral image with 4 or more bands - like Landsat imagery.
```

dcannatella commented on episodes/17-work-with-multi-band-rasters.Rmd:
```diff
@@ -0,0 +1,169 @@
+
+## Create a three-band image
+
+To render a final three band, coloured image in R, we use the `plotRGB()` function.
```
```suggestion
To render a final three-band, coloured image in R, we use the `plotRGB()` function.
```

dcannatella reviewed (commented):
episode 17 reviewed

cforgaci commented:
> Hi, @ClementineCttn. Thanks for the comments.
> 
> > Hello. This lesson is still quite long and a bit abstract. I think it might help to skip a few operations
> 
> * [x]  Drop a few operations to shorten raster lesson

I shortened the lesson as followed:
- removed the hillshade part entirely
- removed one challenge
- simplified a couple of challenges
- removed the mention and figure of CHM in episode 15

cforgaci commented on episodes/15-reproject-raster-data.Rmd:
```diff
@@ -0,0 +1,226 @@
+
+We’ll be looking at another model (the canopy/building height model) in [a later episode](../episodes/16-raster-calculations.Rmd) and will see how to calculate the CHM from the DSM and DTM. Here, we will create a map of the Digital Terrain Model (`DTM_TUD`) of TU Delft and its surrounding draped (or layered) on top of the hillshade (`DTM_hill_TUD`). 
+
+![Source: National Ecological Observatory Network (NEON).](https://datacarpentry.org/r-raster-vector-geospatial/fig/dc-spatial-raster/lidarTree-height.png)
```
I removed this part from episode 15

cforgaci commented:
Thanks all for the input. This lesson is ready to be merged into main.

manuGil commented:
I made a figure illustrating the digital canopy model. I thinks this illustrates it in a slightly better way. **What do you think? How can I improve on it?**

![chm](https://github.com/carpentries-incubator/r-geospatial-urban/assets/8195424/7ac272cc-54b0-441e-9e35-9474da2d684f)

cforgaci commented:
> Hello. This lesson is still quite long and a bit abstract. I think it might help to skip a few operations and instead give some context on why we might want to subtract raster or what to do with the result (in the context of urban and social research and practice)

Thanks @manuGil for taking the time to improve this image! It is clearer, but the issue is that the terrain is still kept in the CHM. The subtraction should lead to a 'flat' ground so that only the height of trees and buildings is visible.

Let's keep the original illustration for now. I will open an issue in the main repo for us to remember to update the figure after our upcoming workshop.
